store/store.go

```diff
@@ -369,51 +369,52 @@ func authenticateUser(r *http.Request, user *auth.UserState) {
 }
 
 // refreshMacaroon will request a refreshed discharge macaroon for the user
-func refreshMacaroon(user *auth.UserState) error {
+func refreshMacaroon(user *auth.UserState) ([]string, error) {
+	newDischarges := make([]string, len(user.StoreDischarges))
 	for i, d := range user.StoreDischarges {
 		discharge, err := MacaroonDeserialize(d)
 		if err != nil {
-			return err
+			return nil, err
 		}
-		if discharge.Location() == UbuntuoneLocation {
-			refreshedDischarge, err := RefreshDischargeMacaroon(d)
-			if err != nil {
-				return err
-			}
-			user.StoreDischarges[i] = refreshedDischarge
+		if discharge.Location() != UbuntuoneLocation {
+			newDischarges[i] = d
+			continue
 		}
+
+		refreshedDischarge, err := RefreshDischargeMacaroon(d)
+		if err != nil {
+			return nil, err
+		}
+		newDischarges[i] = refreshedDischarge
 	}
-	return nil
+	return newDischarges, nil
 }
 
 // refreshUser will refresh user discharge macaroon and update state
 func (s *Store) refreshUser(user *auth.UserState) error {
-	err := refreshMacaroon(user)
+	newDischarges, err := refreshMacaroon(user)
```

Comment:
I think there is a difference here from the original behaviour, since before if authContext was nil, user still got its discharge refreshed, and then the next request should work; with the changes, without an authContext, the user instance won't get the refreshed discharge, correct?
(not necessarily an issue, not sure if there is any case we can have a store instanced without an authContext? )